src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt

```diff
@@ -79,18 +81,47 @@ class SignInWithSourcegraphPanel(private val project: Project) : JPanel() {
             3, ColorUtil.brighter(UIUtil.getPanelBackground(), 3), UIUtil.getPanelBackground())
     separatorPanel.add(separatorComponent)
     panelWithTheMessage.add(separatorPanel)
-    val buttonPanel = JPanel(BorderLayout())
-    buttonPanel.add(signInWithSourcegraphButton, BorderLayout.CENTER)
-    buttonPanel.setOpaque(false)
-    panelWithTheMessage.add(buttonPanel)
+    val buttonPanelGithub = JPanel(BorderLayout())
+    val buttonPanelGitlab = JPanel(BorderLayout())
+    val buttonPanelGoogle = JPanel(BorderLayout())
+    buttonPanelGithub.add(signInWithGithubButton, BorderLayout.CENTER)
+    buttonPanelGitlab.add(signInWithGitlabButton, BorderLayout.CENTER)
+    buttonPanelGoogle.add(signInWithGoogleButton, BorderLayout.CENTER)
+    panelWithTheMessage.add(buttonPanelGithub)
+    panelWithTheMessage.add(buttonPanelGitlab)
+    panelWithTheMessage.add(buttonPanelGoogle)
     setLayout(VerticalFlowLayout(VerticalFlowLayout.TOP, 0, 0, true, false))
     setBorder(JBUI.Borders.empty(PADDING))
     this.add(panelWithTheMessage)
     this.add(createPanelWithSignInWithAnEnterpriseInstance())
   }
 
+  private fun getSignInAction(
+      signInWithGithubButton: JButton,
+      logInToSourcegraphAction: LogInToSourcegraphAction
+  ): (e: ActionEvent) -> Unit {
+    val functionGithub: (e: ActionEvent) -> Unit = {
+      val dataContext = DataManager.getInstance().getDataContext(signInWithGithubButton)
+      val dataContextWrapper = DataContextWrapper(dataContext)
+      val accountsHost: CodyAccountsHost = CodyPersistentAccountsHost(project)
+      dataContextWrapper.putUserData(CodyAccountsHost.KEY, accountsHost)
+      val event =
+          AnActionEvent(
+              null,
+              dataContext,
+              ActionPlaces.POPUP,
+              Presentation(),
+              ActionManager.getInstance(),
+              it.modifiers)
+      if (ActionUtil.lastUpdateAndCheckDumb(logInToSourcegraphAction, event, false)) {
+        ActionUtil.performActionDumbAwareWithCallbacks(logInToSourcegraphAction, event)
+      }
+    }
+    return functionGithub
+  }
+
   fun addMainButtonActionListener(actionListener: ActionListener) {
```

Comment:
so now sign  in with github button is the main button - my question is:
what do we mean by the main button?

Reply:
It is old implementation. TBH i dont know if this is of any use, im just gonna remove it.